packages/sanity/src/desk/comments/src/hooks/useNotificationTarget.ts

```diff
@@ -45,15 +43,25 @@ export function useNotificationTarget(
   const {published, draft} = previewState || {}
   const documentTitle = (draft?.title || published?.title || 'Sanity document') as string
 
-  const currentUrl = new URL(window.location.href)
-  const deskToolSegment = currentUrl.pathname.split('/').slice(2, 3).join('')
-  currentUrl.pathname = `${basePath}/${deskToolSegment}/__edit__${documentId},type=${documentType}`
-  const notificationUrl = currentUrl.toString()
+  const handleGetNotificationValue = useCallback(
+    ({commentId}: {commentId: string}) => {
+      // Generate a path based on the current pane params.
+      // We force a value for `inspect` to ensure that this is included in URLs when comments
+      // are created outside of the inspector context (i.e. directly on the field)
+      // @todo: consider filtering pane router params and culling all non-active RHS panes prior to generating this link
+      const path = createPathWithParams({
```

Comment:
Nice that this function was useful here as well! 👍 

We don't do the "force thing" when copying the comment URL, we should maybe add it there as well.